test/e2e/windows/gmsa_full.go

```diff
@@ -0,0 +1,416 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This test ensures that the whole GMSA process works as intended.
+// However, it does require a pretty heavy weight set up to run correctly;
+// in particular, it does make a number of assumptions about the cluster it
+// runs against:
+//  * there exists a Windows worker node with the agentpool=windowsgmsa label on it
+//  * that node is joined to a working Active Directory domain.
+//  * a GMSA account has been created in that AD domain, and then installed on that
+//    same worker.
+//  * a valid k8s manifest file containing a single CRD definition has been generated using
+//    https://github.com/kubernetes-sigs/windows-gmsa/blob/master/scripts/GenerateCredentialSpecResource.ps1
+//    with the credential specs of that GMSA account, or type GMSACredentialSpec and named gmsa-e2e;
+//    and that manifest file has been written to C:\gmsa\gmsa-cred-spec-gmsa-e2e.yml
+//    on that same worker node.
+//  * the API has both MutatingAdmissionWebhook and ValidatingAdmissionWebhook
+//    admission controllers enabled.
+//  * the cluster comprises at least one Linux node that accepts workloads - it
+//    can be the master, but any other Linux node is fine too. This is needed for
+//    the webhook's pod.
+// All these assumptions are fulfilled by an AKS extension when setting up the AKS
+// cluster we run daily e2e tests against, but they do make running this test
+// outside of that very specific context pretty hard.
+
+package windows
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	"github.com/pkg/errors"
+)
+
+const (
+	// gmsaFullNodeLabel is the label we expect to find on at least one node
+	// that is then expected to fulfill all the expectations explained above.
+	gmsaFullNodeLabel = "agentpool=windowsgmsa"
+
+	// gmsaCrdManifestPath is where we expect to find the manifest file for
+	// the GMSA cred spec on that node - see explanations above.
+	gmsaCrdManifestPath = `C:\gmsa\gmsa-cred-spec-gmsa-e2e.yml`
+
+	// gmsaCustomResourceName is the expected name of the GMSA custom resource
+	// defined at gmsaCrdManifestPath
+	gmsaCustomResourceName = "gmsa-e2e"
+
+	// gmsaWebhookDeployScriptURL is the URL of the deploy script for the GMSA webook
+	// TODO(wk8): we should pin versions.
+	gmsaWebhookDeployScriptURL = "https://raw.githubusercontent.com/kubernetes-sigs/windows-gmsa/master/admission-webhook/deploy/deploy-gmsa-webhook.sh"
```

Comment:
To be honest, not a fan of this URL being here in the tests. Then again, I don't really have any alternative ideea :/